R/FilteredDataset.R

```diff
@@ -235,16 +235,12 @@ FilteredDataset <- R6::R6Class( # nolint
               width = 4,
               tagList(
                 if (allow_add) {
-                  tags$a(
+                  actionLink(
+                    ns("toggle_add_panel"),
+                    label = "",
+                    icon = icon("plus", lib = "font-awesome"),
                     class = "remove pull-right",
-                    tags$i(
-                      class = "fa fa-plus",
-                      title = "fold/expand transform panel",
-                      onclick = sprintf(
-                        "togglePanelItem(this, '%s', 'fa-minus', 'fa-plus');",
```

Comment:
what is the root problem of `togglePanelItem`?

Reply:
The root problem is that the [shinyjs::toggle](https://github.com/insightsengineering/teal.slice/blob/3707239ec62a3ad2ce9df5037bd75c048598b79f/R/FilteredDataset.R#L346-L348) observer gets loaded as many times as the `srv_active` is called.

```
observeEvent(input$toggle_add_panel, {
  shinyjs::toggle("add_panel")
})
```

Reply:
I think `observeEvent` can be removed when `togglePanelItem` is used. I forgot to remove `observeEvent` after replacing `actionLink` with `tags$a` with js function.

Reply:
Ah! true. Removing it should do it.

Reply:
But, the teal PR is still valid as it fixes other observers not getting re-initialized

Reply:
I still have some issue with the add. Let me test it a little.

Reply:
We can have a call about `ns` of the filter-panel tomorrow with @averissimo . I think we are mixing few issues in the same time

Reply:
I did talk with @averissimo today and the observer destroy issue that he is working on is for the filter state (The filter card observers that need to be removed when filter states are removed). And this is regarding removing observers of the whole filter panel (well the filter panel that we're using)